pycqed/measurement/detector_functions.py

```diff
@@ -965,17 +965,20 @@ def __init__(self, HS, threshold=1.75, trigger_separation=20e-6,
         self.last_frequency = 0.
         self.threshold = threshold
         self.last = 1.
-        self.trigger_separation = trigger_separation
-        self.demod_mode = demod_mode
+        HS.trigger_separation(trigger_separation)
```

Comment:
I think specifying the trigger separation should indeed be done as a parameter (as you've implemented it). However, then we may want to remove it from the detector function. This does mean we have to change it everywhere it is used, but I think this is a good idea.